pkg/virt-handler/hotplug-disk/mount.go

```diff
@@ -174,23 +174,20 @@ func (m *volumeMounter) deleteMountTargetRecord(vmi *v1.VirtualMachineInstance)
 		return fmt.Errorf(unableFindHotplugMountedDir)
 	}
 
-	recordFile := filepath.Join(m.mountStateDir, string(vmi.UID))
-	exists, err := diskutils.FileExists(recordFile)
-	if err != nil {
-		return err
+	record := vmiMountTargetRecord{}
+	err := m.checkpointManager.Get(string(vmi.UID), &record)
+	if err != nil && !errors.Is(err, os.ErrNotExist) {
+		return fmt.Errorf("failed to get checkpoint %s, %w", vmi.UID, err)
 	}
 
-	if exists {
-		record, err := m.getMountTargetRecord(vmi)
-		if err != nil {
-			return err
-		}
-
+	if err == nil {
```

Comment:
Shouldn't it be?:
```
if !errors.Is(err, os.ErrNotExist) {
```

Reply:
So the code goes as following:
Error occurs but it is not `ErrNotExist` then we simply return as we can't do much.
Following it if there was no error, meaning the file/record was there, then we can actually use the record to remove the paths.
If there would be `ErrNotExist` then the record is already gone and we can't do much then fix the cache.

Reply:
Makes sense, but for [containerdisks](https://github.com/kubevirt/kubevirt/pull/10848/commits/25aee423f7321d8ab5f5cc1573e52ee6011a15e2) we perform a different check. 
So  maybe we should do the same there

Reply:
Logic is the same but nevertheless I aligned it in container disk to match the hotplug